### PR TITLE
feat(controlplane): add overriding the mailing options

### PR DESCRIPTION
**Pull request by AndreasZeissner**
## Motivation and Context

This is a follow up to: https://github.com/wundergraph/cosmo/pull/617

It introduces overriding the smtp server in the controlplane by introducing the following environment variables:

`SMTP_HOST:` the host to connect to (no default)   
`SMTP_PORT:`  the port to use (defaults to 587)
`SMTP_SECURE:` enable or disable smtps (defaults to true)
`SMTP_REQUIRE_TLS:` enable or disable using TLS (defaults to true)

## TODO

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

File: helm/cosmo/charts/controlplane/templates/deployment.yaml [hunk 1]
```diff
@@ -236,6 +236,30 @@ spec:
                 secretKeyRef:
                   name: {{ include "controlplane.secretName" . }}
                   key: smtpUsername
+            - name: SMTP_PORT
```
Comment (StarpTech) [inline helm/cosmo/charts/controlplane/templates/deployment.yaml:239]: Ideally, we set only add variables when the value is set.

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: helm/cosmo/charts/controlplane/templates/deployment.yaml (hunk 1 above)
Comment (StarpTech) [reply helm/cosmo/charts/controlplane/templates/deployment.yaml:239]: Sounds good.

File: controlplane/src/core/env.schema.ts [hunk 2]
```diff
@@ -105,8 +105,26 @@ export const envVariables = z
...
+      .string()
+      .transform((val) => val === 'true')
+      .default('false'),
     SMTP_USERNAME: z.string().optional(),
     SMTP_PASSWORD: z.string().optional(),
+    SMTP_HOST: z.string().optional(),
+    SMTP_PORT: z
+      .string()
+      .optional()
+      .default('') // here to avoid possible type of number | "" | undefined type
```
Comment (AndreasZeissner) [inline controlplane/src/core/env.schema.ts:118]: @StarpTech you might have a better solution for this.

If I leave away the default, tsc interferes `number | "" | undefined` as possible type as I would have to do something like this:

```
.transform((val) => val && Number.parseInt(val)),     
```

As val has then no explicit type. I could also handle it throughout the interfaces, but it feels bad, as all other types are usually `undefined | string` or `undefined | number` or such.

Comment (StarpTech) [reply controlplane/src/core/env.schema.ts:118]: Do we need a default?

**Review by StarpTech — APPROVED** [review]
LGTM

File: controlplane/src/core/env.schema.ts (hunk 2 above)
Comment (StarpTech) [reply controlplane/src/core/env.schema.ts:118]:
```ts
      SMTP_PORT: z
          .coerce.number().optional()
```

Comment (AndreasZeissner) [reply controlplane/src/core/env.schema.ts:118]: Thank you, this will now avoid the default.

**Pull request merged**